### PR TITLE
Mip 242 - Adding Integration Test cases

### PR DESCRIPTION
**Pull request by manoj-github-avio**

### CODE REVIEW

File: demo/README-demo.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+### Executing get-certificate-flow
+
+Execute the following to test the Get Certificate component
+
+curl GET 'http://localhost:8081/api/certificate'
+
+### Note
+
+pass the clientId, ClientSecret and tenantId as arguments while deploying the application
```
Comment (kkingavio) [inline demo/README-demo.md:37]: might want to expand on this... also, the demo app looks for a certain key/secret/cert.... you should mention those as well... or set them up as parameters...

File: src/test/resources/integration-test-mule-config.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<mule xmlns="http://www.mulesoft.org/schema/mule/core"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:akv="http://www.mulesoft.org/schema/mule/akv"
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+          http://www.mulesoft.org/schema/mule/akv http://www.mulesoft.org/schema/mule/akv/current/mule-akv.xsd">
+
+    <akv:config name="config" >
+        <akv:connection vaultName="akv-mule-integration-key" tenantId="cb305f69-a885-4c50-8f0e-1aa14f9b4fa0" clientId="ba570879-dd92-4560-9a10-594f3bc62793" clientSecret="6Z7C8Smx631By.a01F.YTt4-M993H_c4-6" />
```
Comment (kkingavio) [inline src/test/resources/integration-test-mule-config.xml:8]: shouldn't have these values in here... use system parameters (like keyName)

File: demo/README-demo.md (hunk 1 above)
Comment (manoj-github-avio) [reply demo/README-demo.md:38]: added as parameters, added mvn command to run tests and resolving

File: src/test/resources/integration-test-mule-config.xml (hunk 2 above)
Comment (manoj-github-avio) [reply src/test/resources/integration-test-mule-config.xml]: changed to system params

**Comment by kkingavio** [comment]
Looks good to me

File: src/test/java/akv/AkvIntegrationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    @Override
+    protected String getConfigFile() {
+        return "integration-test-mule-config.xml";
+    }
+
+    private static void setGlobals(){
+        //System.setProperty("AZURE_CLIENT_ID", "");
+        //System.setProperty("AZURE_CLIENT_SECRET", "");
+        //System.setProperty("AZURE_TENANT_ID", "");
+        System.setProperty("AKV_TEST_KEY_NAME", getKeyName());
```
Comment (adesjardin) [inline src/test/java/akv/AkvIntegrationTestCase.java:52]: Normally environment variables use this format but java system properties are usually lower case with . notation such as akv.test.key.name.  We should follow those commonly used standards.

File: pom.xml [hunk 4]
```diff
@@ -103,6 +127,12 @@
             <artifactId>mockserver-client-java</artifactId>
             <version>5.7.0</version>
         </dependency>
+        <dependency>
+            <groupId>com.azure</groupId>
+            <artifactId>azure-identity</artifactId>
+            <version>1.0.4</version>
```
[the remark this answers is not in the document]
Comment (manoj-github-avio) [reply pom.xml:133]: moved version to property

File: src/test/java/akv/AkvIntegrationTestCase.java (hunk 3 above)
Comment (manoj-github-avio) [reply src/test/java/akv/AkvIntegrationTestCase.java:52]: made changes to accept lowercase dot notation

**Pull request merged**